pilot/pkg/model/authorization.go

```diff
@@ -76,32 +85,52 @@ func (policy *AuthorizationPolicies) ListAuthorizationPolicies(namespace string,
 	if policy.RootNamespace != "" {
 		namespaces = append(namespaces, policy.RootNamespace)
 	}
-	// To prevent duplicate policies in case root namespace equals proxy's namespace.
-	if namespace != policy.RootNamespace {
-		namespaces = append(namespaces, namespace)
+
+	// Policies can be in root namespace and have workload selectors
+	// This prevent duplicate policies in case root namespace equals proxy's namespace.
+	if proxyInfo.Namespace != policy.RootNamespace {
+		namespaces = append(namespaces, proxyInfo.Namespace)
```

Comment:
I am concerned that the current logic of looping through root ns and workload ns may end up applying a policy in the root ns for a waypoint that isn't in the root ns. 

Do I need to prevent this here? _By refactoring waypoint logic out of the ns loop or adding a check for if the root ns is not equal to the waypoint ns._ @keithmattix

Reply:
Yeah waypoints don't care if a policy exists in the root namespace. I believe L85-87 shouldn't apply to waypoints